packages/code-gen/src/generator/openAPI/generator.js

```diff
@@ -89,29 +87,28 @@ export function generateOpenApiFile(structure, options) {
         tags: [route.group],
         description: route.docString,
         operationId: route.uniqueName,
-        // query, params
-        ...transformParams(structure, route),
-        // requestBody (with files, if any)
-        ...transformBody(structure, route),
-        responses: constructResponse(structure, route),
+        ...transformParams(structure, route, uniqueNameSet),
+        ...transformBody(structure, route, uniqueNameSet),
+        responses: constructResponse(structure, route, uniqueNameSet),
+        // @ts-ignore
+        ...(options.openApiRouteExtensions?.[route.uniqueName] ?? {}),
       };
     }
+  }
 
-    /**
-     * @type {import("../../generated/common/types").CodeGenType[]}
-     */
-    // @ts-ignore
-    const groupComponents = Object.values(groupStructure).filter(
-      (it) => it.type !== "route",
-    );
+  const flattenStructure = Object.assign(
+    {},
+    ...Object.values(structure).flat(),
+  );
```

Comment:
If I understand this correctly, this wont work if names are used in multiple groups, eg;

```
new TypeCreator("user").router("/user").get("/", "list"); // -> User List
new TypeCreator("todo").router("/user").get("/", "list"); // -> Todo List
```

Reply:
Oke good catch! Don't know why, but I assumed/thought that the uniqueName/key was used as identifier.